generators/client/templates/angular/src/main/webapp/app/admin/audits/audits.component.ts.ejs

```diff
@@ -30,113 +29,109 @@ import { AuditsService } from './audits.service';
   selector: '<%= jhiPrefixDashed %>-audit',
   templateUrl: './audits.component.html'
 })
-export class AuditsComponent implements OnInit, OnDestroy {
-    audits: Audit[];
-    fromDate: string;
-    itemsPerPage: any;
-    links: any;
-    page: number;
-    routeData: any;
-    predicate: any;
-    previousPage: any;
-    reverse: boolean;
-    toDate: string;
-    totalItems: number;
+export class AuditsComponent implements OnInit {
+    private dateFormat = 'yyyy-MM-dd';
+    audits: Audit[] = [];
+    fromDate = '';
+    itemsPerPage = ITEMS_PER_PAGE;
+    page!: number;
+    predicate!: string;
+    previousPage!: number;
+    ascending!: boolean;
+    toDate = '';
+    totalItems = 0;
+    canLoad = false;
+    initialized = false;
+    dataHaveBeenLoaded = false;
 
     constructor(
         private auditsService: AuditsService,
-        private alertService: JhiAlertService,
-        private parseLinks: JhiParseLinks,
         private activatedRoute: ActivatedRoute,
         private datePipe: DatePipe,
         private router: Router
-    ) {
-        this.itemsPerPage = ITEMS_PER_PAGE;
-        this.routeData = this.activatedRoute.data.subscribe((data) => {
+    ) {}
+
+    ngOnInit(): void {
+        this.today();
+        this.previousMonth();
+        this.activatedRoute.data.subscribe(data => {
             this.page = data['pagingParams'].page;
             this.previousPage = data['pagingParams'].page;
-            this.reverse = data['pagingParams'].ascending;
+            this.ascending = data['pagingParams'].ascending;
             this.predicate = data['pagingParams'].predicate;
+            this.canLoad = true;
+            this.loadData();
         });
     }
 
-    ngOnInit() {
-        this.today();
-        this.previousMonth();
-        this.loadAll();
-    }
-
-    ngOnDestroy() {
-        this.routeData.unsubscribe();
-    }
-
-    previousMonth() {
-        const dateFormat = 'yyyy-MM-dd';
-        let fromDate: Date = new Date();
-
-        if (fromDate.getMonth() === 0) {
-            fromDate = new Date(fromDate.getFullYear() - 1, 11, fromDate.getDate());
+    private previousMonth(): void {
+        let date = new Date();
+        if (date.getMonth() === 0) {
+            date = new Date(date.getFullYear() - 1, 11, date.getDate());
         } else {
-            fromDate = new Date(fromDate.getFullYear(), fromDate.getMonth() - 1, fromDate.getDate());
+            date = new Date(date.getFullYear(), date.getMonth() - 1, date.getDate());
         }
-
-        this.fromDate = this.datePipe.transform(fromDate, dateFormat);
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.fromDate = this.datePipe.transform(date, this.dateFormat)!;
     }
 
-    today() {
-        const dateFormat = 'yyyy-MM-dd';
+    private today(): void {
```

Comment:
I would prefer this function returning today + 1 Date and then you can do this.toDate = this.today()
WDYT ?

Reply:
Good idea, done and made the same also for the `previousMonth()` function.